### PR TITLE
Introduce component state #13

### PR DESCRIPTION
**Pull request by Ayselin**
Introduce component state.

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,23 +1,24 @@
 .App {
   text-align: center;
+  
 }
 
 .container {
   display: grid;
  grid-template-columns: repeat(auto-fit, minmax(400px, 1fr));
-  margin: 30px;
+
```
Comment (rmallols) [inline src/App.css]: Empty line :)

File: src/App.css [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
   padding: 20px; 
-  border: 1px solid green;
   position: relative;
   overflow: hidden;
   
 }
 
 img {
   width: 100%;
+
```
Comment (rmallols) [inline src/App.css]: Empty line :)

File: src/Details.js [hunk 3]
```diff
@@ -1,8 +1,30 @@
...
+
+  componentDidMount() {
+     setTimeout(() => {
+        this.setState({
+          welcomeMessage:'The best is yet to come!!!!'
+        });     
+     },3000);
+  }
+
+
```
Comment (rmallols) [inline src/Details.js]: Empty line :)

**Review by rmallols — APPROVED** [review]
Looks good Aysel - please get rid of the unnecessary empty lines before merging!

**Comment by Ayselin** [comment]
Will do Ricardo!

Thanks :)


________________________________

**Review by rmallols — APPROVED** [review]

File: src/App.js [hunk 4]
```diff
@@ -1,25 +1,26 @@
 import React, { Component } from 'react';
 import Gallery from './Gallery';
-import { BrowserRouter as Router, Route } from 'react-router-dom';
-import Details from './Details.js';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import Details from './Details/Details.js';
 import './App.css';
-
+import NotExist from './NotExist.js';
```
Comment (rmallols) [inline src/App.js:8]: Unnecessary empty line here

File: src/Details/Details.css [hunk 5]
```diff
@@ -0,0 +1,5 @@
+.Gallery {
+   width: 50%;
+   height: 50%;
+
```
Comment (rmallols) [inline src/Details/Details.css:4]: Empty line here

File: src/Details/Details.css [hunk 6]
```diff
@@ -0,0 +1,5 @@
+.Gallery {
```
Comment (rmallols) [inline src/Details/Details.css:1]: Mmm...why the `Details` component starts with a `Gallery` class? it should be `Details` instead for consistency reasons!

File: src/Details/Details.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { Link, Redirect } from 'react-router-dom';
+import getGallery from '../GetGallery-get';
+import './Details.css';
+
+
```
Comment (rmallols) [inline src/Details/Details.js:6]: Empty line here :)

File: src/Details/Details.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  componentDidMount() {
+    let galleryId = this.props.match.params.galleryId;
+    let gallery = getGallery()
+      .find((gallery) => gallery.id === galleryId);
+    this.setState({
+      gallery: gallery
+    });
+
```
Comment (rmallols) [inline src/Details/Details.js:23]: Emtpy line

**Review by rmallols — CHANGES_REQUESTED** [review]

**Pull request merged**